cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -168,17 +168,26 @@ class ScalarAggregateNode : public ExecNode {
 
   Status DoConsume(const ExecBatch& batch, size_t thread_index) {
     util::tracing::Span span;
-    START_SPAN(span, "Consume",
-               {{"aggregate", ToStringExtra()},
-                {"node.label", label()},
-                {"batch.length", batch.length}});
+    START_SPAN(
+        span, "Consume",
+        {{"aggregate", ToStringExtra()},
+         {"node.label", label()},
+         {"batch.length", batch.length},
+         {"memory_pool_bytes",
+          plan_->exec_context()->memory_pool()->bytes_allocated()},
+         {"memory_used", arrow::internal::tracing::GetMemoryUsed()},
+         {"memory_used_process", arrow::internal::tracing::GetMemoryUsedByProcess()}});
```

Comment:
Background thread on a fixed schedule was the original ask but I think there was some concern around having two streams of telemetry information.  As a start I think only using `memory_pool_bytes` would be sufficient.  Longer term it could be handy to have some idea of the RSS of the process just to have some sense for how we are doing on fragmentation and non-pool allocations.  Since a system call is expensive I wonder if there would be a way to debounce this information and only report it at most every 10ms or on some configurable threshold.

Reply:
@westonpace I can create the background thread for report the `system RAM`, but where can I place it? inside the memory pool constructor?  and using the same exporter or creating a new one just for this.

Reply:
To be clear, I am not asking for a background thread in that comment.  It is ok either way.  I have no preference either way.  I only mentioned the background thread in the comment to explain the context.

In both cases (background thread or span processor) I now think we want to move the initialization into the query testing tool (which unfortunately, is not yet merged).  This means the feature will not be enabled for library users but I think that is ok.